software/apps/audio_module/event_detector/Makefile

```diff
@@ -0,0 +1,17 @@
+# makefile for user application
+
+# the current directory
+APP_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+# files needed for this code
+C_SRCS   := $(wildcard *.c) 
```

Comment:
This is the Makefile variable you want to adjust if you want to have a non-flat hierarchy. I believe something like this would include all C files in all subdirectories work:

```
C_SRCS := $(wildcard *.c)
C_SRCS += $(wildcard **/*.c)
```

You could also be more explicit if you only want to include certain subdirectories or files.